pymodbus/constants.py

```diff
@@ -3,155 +3,157 @@
 This is the single location for storing default
 values for the servers and clients.
 """
+import enum
+
 
 INTERNAL_ERROR = "Pymodbus internal error"
 
 
-class ModbusStatus:  # pylint: disable=too-few-public-methods
+class ModbusStatus(int, enum.Enum):
     """These represent various status codes in the modbus protocol.
 
-    .. attribute:: Waiting
+    .. attribute:: WAITING
 
        This indicates that a modbus device is currently
        waiting for a given request to finish some running task.
 
-    .. attribute:: Ready
+    .. attribute:: READY
 
        This indicates that a modbus device is currently
        free to perform the next request task.
 
-    .. attribute:: On
+    .. attribute:: ON
 
        This indicates that the given modbus entity is on
 
-    .. attribute:: Off
+    .. attribute:: OFF
 
        This indicates that the given modbus entity is off
 
-    .. attribute:: SlaveOn
+    .. attribute:: SLAVE_ON
 
        This indicates that the given modbus slave is running
 
-    .. attribute:: SlaveOff
+    .. attribute:: SLAVE_OFF
 
        This indicates that the given modbus slave is not running
     """
 
-    Waiting = 0xFFFF
-    Ready = 0x0000
-    On = 0xFF00
-    Off = 0x0000
-    SlaveOn = 0xFF
-    SlaveOff = 0x00
+    WAITING = 0xFFFF
+    READY = 0x0000
+    ON = 0xFF00
+    OFF = 0x0000
+    SLAVE_ON = 0xFF
+    SLAVE_OFF = 0x00
 
-    def __init__(self):
```

Comment:
Why should we permit objects ? I do not see the need in contrary.

Reply:
Enum members are singletons by default, we don't need any special logic to guard them, Python handles that for us.

Reply:
It does not prohibit that I can assign the class to a variable and make changes, with unexpected results.

Reply:
Not in the case of enums, they are using metaclasses underneath. Say, if somebody does `ModbusStatus(0xFFFF)`, they will get `ModbusStatus.WAITING`.